compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/WindowComposeBridge.desktop.kt

```diff
@@ -76,17 +76,33 @@ internal class WindowComposeBridge(
     override val renderApi: GraphicsApi
         get() = component.renderApi
 
+    override val interopBlendingSupported: Boolean
+        get() = when(renderApi) {
+            GraphicsApi.DIRECT3D, GraphicsApi.METAL -> true
+            else -> false
+        }
+
     override val clipComponents: MutableList<ClipRectangle>
         get() = component.clipComponents
 
     override val focusComponentDelegate: Component
         get() = component.canvas
 
-    internal var transparency: Boolean
+    var transparency: Boolean
         get() = component.transparency
         set(value) {
             component.transparency = value
-            platformContext.isWindowTransparent = value
+            if (value && !isWindowTransparent && renderApi == GraphicsApi.METAL) {
+                /*
+                 * SkiaLayer sets background inside transparency setter, that is required for
+                 * cases like software rendering.
+                 * In case of transparent Metal canvas on opaque window, background values with
+                 * alpha == 0 will make the result color black after clearing the canvas.
+                 *
+                 * Reset it to null to keep the color default.
+                 */
+                component.background = null
```

Comment:
This logic seems should belong to `SkiaLayer`. Can we move it there?

I.e. can we write something like this?
```
    actual var transparency: Boolean
            ...
            if (!value) {
                ...
            } else {
                background = null
            }
        }
```
or
```
    actual var transparency: Boolean
            ...
            if (!value) {
                ...
            } else {
                background = if (renderApi == GraphicsApi.METAL) null else Color(0, 0, 0, 0)
            }
        }
```
without checking `isWindowTransparent`, is it mandatory to check it?

Reply:
> without checking isWindowTransparent, is it mandatory to check it?

Yes it's why this is here. It's explained in the comment:
```
In case of transparent Metal canvas on opaque window, background values with
alpha == 0 will make the result color black after clearing the canvas.
```

Reply:
🫠 okay